File: server/com.eclipsesource.uml.glsp/src/main/java/com/eclipsesource/uml/glsp/uml/deployment_diagram/constants/DeploymentTypes.java
/********************************************************************************
 * Copyright (c) 2022 EclipseSource and others.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License v. 2.0 which is available at
 * https://www.eclipse.org/legal/epl-2.0, or the MIT License which is
 * available at https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: EPL-2.0 OR MIT
 ********************************************************************************/
package com.eclipsesource.uml.glsp.uml.deployment_diagram.constants;

import com.eclipsesource.uml.glsp.utils.UmlConfig;

public final class DeploymentTypes {

   // DEPLOYMENT DIAGRAM
   public static final String ICON_DEPLOYMENT_NODE = UmlConfig.Types.PRE_ICON + "node";
   public static final String DEPLOYMENT_NODE = UmlConfig.Types.PRE_NODE + "node";
   public static final String LABEL_NODE_NAME = UmlConfig.Types.PRE_LABEL + "node:name";
   public static final String ICON_ARTIFACT = UmlConfig.Types.PRE_ICON + "artifact";
   public static final String ARTIFACT = UmlConfig.Types.PRE_NODE + "artifact";
   public static final String COMMUNICATION_PATH = UmlConfig.Types.PRE_EDGE + "communicationpath";
   public static final String DEPLOYMENT = UmlConfig.Types.PRE_EDGE + "deployment";
   public static final String ICON_EXECUTION_ENVIRONMENT = UmlConfig.Types.PRE_ICON + "executionenvironment";
   public static final String EXECUTION_ENVIRONMENT = UmlConfig.Types.PRE_NODE + "executionenvironment";
   public static final String ICON_DEVICE = UmlConfig.Types.PRE_ICON + "device";
   public static final String DEVICE = UmlConfig.Types.PRE_NODE + "device";
   public static final String ICON_DEPLOYMENT_SPECIFICATION = UmlConfig.Types.PRE_ICON + "deploymentspecification";
   public static final String DEPLOYMENT_SPECIFICATION = UmlConfig.Types.PRE_NODE + "deploymentspecification";
   public static final String DEPLOYMENT_COMPONENT = UmlConfig.Types.PRE_NODE + "deploymentcomponent";

   // COMMON CANDIDATE
   public static final String STRUCTURE = "struct";

   // SHARED WITH CLASS AND USECASE
   public static final String LABEL_EDGE_MULTIPLICITY = UmlConfig.Types.PRE_LABEL + "edge-multiplicity";

   private DeploymentTypes() {}
}

File: client/packages/uml-glsp/src/uml/diagram/deployment/deployment.types.ts
/********************************************************************************
 * Copyright (c) 2023 EclipseSource and others.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License v. 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0.
 *
 * This Source Code may also be made available under the following Secondary
 * Licenses when the conditions for such availability set forth in the Eclipse
 * Public License v. 2.0 are satisfied: GNU General Public License, version 2
 * with the GNU Classpath Exception which is available at
 * https://www.gnu.org/software/classpath/license.html.
 *
 * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
 ********************************************************************************/
import {UmlDiagramType} from '@borkdominik-biguml/uml-common';
import {DefaultTypes} from '@eclipse-glsp/client';
import {QualifiedUtil} from '../../qualified.utils';

export namespace UmlDeploymentTypes {
    export const DEPLOYMENT_SPECIFICATION = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.NODE, 'DeploymentSpecification');

    export const ARTIFACT = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.NODE, 'Artifact');

    export const DEVICE = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.NODE, 'Device');

    export const EXECUTION_ENVIRONMENT = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.NODE, 'ExecutionEnvironment');

    export const MODEL = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.NODE, 'Model');

    export const NODE = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.NODE, 'Node');

    export const PACKAGE = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.NODE, 'Package');

    export const COMMUNICATION_PATH = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.EDGE, 'CommunicationPath');
    export const DEPENDENCY = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.EDGE, 'Dependency');
    export const MANIFESTATION = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.EDGE, 'Manifestation');
    export const DEPLOYMENT = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.EDGE, 'Deployment');

    export const PROPERTY_LABEL_TYPE = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.LABEL, 'Property-type');

    export const PROPERTY_LABEL_MULTIPLICITY = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.LABEL, 'Property-multiplicity');

    export const PROPERTY = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.COMPARTMENT, 'Property');

    export const OPERATION = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.COMPARTMENT, 'Operation');

    export const GENERALIZATION = QualifiedUtil.representationTypeId(UmlDiagramType.DEPLOYMENT, DefaultTypes.EDGE, 'Generalization');

}
